tex/gregoriotex.lua

```diff
@@ -120,6 +120,37 @@ local translation_mark = 1
 local abovelinestext_mark = 2
 log("marker whatsit id is %d", marker_whatsit_id)
 
+local function gregorio_exe()
+  if real_gregorio_exe == nil then
+    local exe_version
+
+    -- first look for one with the exact version
+    real_gregorio_exe = 'gregorio-5_0_0-rc1' -- FILENAME_VERSION
+    exe_version = io.popen(real_gregorio_exe..' --version', 'r')
+    exe_version = exe_version:read("*line")
+    if not exe_version then
+      -- look for suffix-less executable
+      real_gregorio_exe = 'gregorio'
+      exe_version = io.popen(real_gregorio_exe..' --version', 'r')
+      exe_version = exe_version:read("*line")
+      if not exe_version or string.match(exe_version,"%d+%.%d+%.")
+          ~= string.match(internalversion,"%d+%.%d+%.") then
+        real_gregorio_exe = nil
```

Comment:
Calling `err` will generate a TeX error, interrupting the run, so we would not check again unless the user (in interactive mode) instructed the engine to continue, in which case we may want to try again just in case the user has corrected something.  At least that was my thinking, but I can change it.